src/interrupt/trigger/eventfd.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2021 Amazon.com, Inc. or its affiliates.
+// All Rights Reserved.
+
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Implementation for a Trigger based on EventFd.
+
+use std::io;
+use std::ops::Deref;
+use vmm_sys_util::eventfd::EventFd;
+
+use crate::interrupt::trigger::Trigger;
+
+/// Trait used for creating an EventFd from an object.
```

Comment:
Why is this needed? Intuitively you should be able to use the deref implementation to get a clone of the inner `EventFd`. The name of this trait is not appropriate for the implementation because `as` suggests a conversion from one type to another, but the implementation is actually creating a clone.

Here's an example:
```rust
let ev_trigger = EventFdTrigger::new(EventFd::new(0).unwrap());
let inner: &EventFd = ev_trigger.deref();
```

Reply:
I've removed EventFdTrigger along with Trigger based on #71 (comment).